### PR TITLE
Fix for  https://github.com/Microsoft/vso-agent-tasks/issues/1287

### PR DESCRIPTION
**Pull request by mvvsubbu**
https://github.com/Microsoft/vso-agent-tasks/issues/1287

### CODE REVIEW

File: Tasks/IISWebAppDeployment/MsDeployOnTargetMachines.ps1 [hunk 1]
```diff
@@ -212,7 +214,7 @@ function Does-WebSiteExists
         return $true
     }
     
-    Write-Verbose "WebSite does not exist" -Verbose
+    Write-Verbose "Website ($siteName) does not exist" -Verbose
```
Comment (KrishnaAdityaB) [inline Tasks/IISWebAppDeployment/MsDeployOnTargetMachines.ps1]: why ()? did you want it to be Website '$siteName'?

File: Tasks/IISWebAppDeployment/MsDeployOnTargetMachines.ps1 [hunk 2]
```diff
@@ -611,6 +613,14 @@ function Execute-Main
             Enable-SNI -siteName $WebSiteName -sni $ServerNameIndication -ipAddress $IpAddress -port $Port -hostname $HostName
         }
     }
+    else
+    {
+        $doesWebSiteExists = Does-WebSiteExists -siteName $WebSiteName
+        if (-not $doesWebSiteExists)  
+        {
+            throw "The web site '$WebSiteName' does not exist and you did not request to create it - deployment cannot continue."
```
Comment (KrishnaAdityaB) [inline Tasks/IISWebAppDeployment/MsDeployOnTargetMachines.ps1]: 'web site' or 'website' ?

File: Tasks/IISWebAppDeployment/MsDeployOnTargetMachines.ps1 [hunk 3]
```diff
@@ -594,12 +596,12 @@ function Execute-Main
 {
     Write-Verbose "Entering Execute-Main function" -Verbose
 
-    if(-not (IsInputNullOrEmpty -str $AppPoolName))
+    if($createAppPool -ieq "true")
```
Comment (KrishnaAdityaB) [inline Tasks/IISWebAppDeployment/MsDeployOnTargetMachines.ps1]: is this check required here too?
isn't it duplicate? can we consider this private and assume it's not null - public method always do the check and passes it non null value?

**Comment by KrishnaAdityaB** [comment]
looks good

File: Tasks/IISWebAppDeployment/MsDeployOnTargetMachines.ps1 (hunk 2 above)
Comment (mvvsubbu) [inline Tasks/IISWebAppDeployment/MsDeployOnTargetMachines.ps1]: Fixed

File: Tasks/IISWebAppDeployment/MsDeployOnTargetMachines.ps1 (hunk 1 above)
Comment (mvvsubbu) [inline Tasks/IISWebAppDeployment/MsDeployOnTargetMachines.ps1]: I wanted to print "Website (xyz) does not exist" as error message

**Pull request merged**